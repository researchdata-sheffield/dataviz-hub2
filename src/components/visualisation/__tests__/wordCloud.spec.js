/**
 * @jest-environment jsdom
 */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from "react";
import { render, getByText } from "@testing-library/react";
import WordCloud from "../wordCloud";

describe("Visualisation | Word Cloud", () => {
  const words = [
    {
      name: "Test Category 1",
      count: 2,
      type: "category"
    },
    {
      name: "Test Category 2",
      count: 8,
      type: "category"
    },
    {
      name: "Test Tag 1",
      count: 1,
      type: "tag"
    }
  ];

  beforeEach(() => {});

  it("should renders correctly", async () => {
    let wordCloudComponent = render(
      <WordCloud words={words} colours={["white"]} />
    );
    expect(wordCloudComponent).toMatchSnapshot();
    expect(
      await getByText(wordCloudComponent.container, "Test Category 1")
    ).toBeInTheDocument();
    expect(
      await getByText(wordCloudComponent.container, "Test Category 2")
    ).toBeInTheDocument();
    expect(
      await getByText(wordCloudComponent.container, "Test Tag 1")
    ).toBeInTheDocument();
  });
});
