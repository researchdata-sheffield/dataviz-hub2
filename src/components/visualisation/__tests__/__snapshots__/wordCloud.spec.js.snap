// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualisation | Word Cloud should renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKwDye iNUPXw"
      >
        <div
          class="word-cloud-wrap space-y-0 space-x-1"
        >
          <a
            class="word font-semibold"
            gatsby="true"
            href="/visualisation/category/test-category-1"
            style="font-size: 1.5rem; color: white;"
          >
            Test Category 1
          </a>
          <a
            class="word font-semibold"
            gatsby="true"
            href="/visualisation/category/test-category-2"
            style="font-size: 2.1rem; color: white;"
          >
            Test Category 2
          </a>
          <a
            class="word "
            gatsby="true"
            href="/visualisation/tag/test-tag-1"
            style="font-size: 0.9rem; color: white;"
          >
            Test Tag 1
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKwDye iNUPXw"
    >
      <div
        class="word-cloud-wrap space-y-0 space-x-1"
      >
        <a
          class="word font-semibold"
          gatsby="true"
          href="/visualisation/category/test-category-1"
          style="font-size: 1.5rem; color: white;"
        >
          Test Category 1
        </a>
        <a
          class="word font-semibold"
          gatsby="true"
          href="/visualisation/category/test-category-2"
          style="font-size: 2.1rem; color: white;"
        >
          Test Category 2
        </a>
        <a
          class="word "
          gatsby="true"
          href="/visualisation/tag/test-tag-1"
          style="font-size: 0.9rem; color: white;"
        >
          Test Tag 1
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
